### PR TITLE
Allocations fix for diagnostic edmf

### PR DESCRIPTION
**Pull request by nefrathenrici**
This PR:
- fixes allocations for `set_diagnostic_edmf_precomputed_quantities!`
- Adds a performance test for job diagnostic_edmfx_aquaplanet

### CODE REVIEW

**Comment by charleskawczynski** [comment]
Hi @nefrathenrici, this allocation very well may be due to `^2`, see https://github.com/CliMA/ClimaCore.jl/issues/1126

**Review by charleskawczynski — APPROVED** [review]

File: src/cache/diagnostic_edmf_precomputed_quantities.jl [hunk 1]
```diff
@@ -314,31 +325,52 @@ function set_diagnostic_edmf_precomputed_quantities!(Y, p, turbconv_model)
...
                         entr_detrʲ_prev_level.entr * u³ʲ_data_prev_halflevel
                     )
                 )
-
-            @. u³ʲ_halflevel = ifelse(
-                (
-                    u³ʲ_datau³ʲ_data < (FT(1e-6) / ∂x³∂ξ³_level^2) ||
+            # Helper function for broadcasting ternary operator across fields.
+            # This is necessary to prevent allocations using ifelse().
+            function pointwise_ifelse(
```
Comment (charleskawczynski) [inline src/cache/diagnostic_edmf_precomputed_quantities.jl:342]: Is this needed? I haven't seen `ifelse` allocate in the past

Comment (charleskawczynski) [reply src/cache/diagnostic_edmf_precomputed_quantities.jl:342]: Sorry, I don't remember seeing these changes when I approved the PR, this looks highly specialized and it's not clear to me that it's needed

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -843,6 +843,12 @@ steps:
         agents:
           slurm_mem: 20GB
 
+      - label: ":fire: Flame graph: perf target (diagnostic edmfx)"
+        command: "julia --color=yes --project=perf perf/flame.jl --job_id flame_perf_target_diagnostic_edmfx --target_job diagnostic_edmfx_aquaplanet"
+        artifact_paths: "flame_perf_target_diagnostic_edmfx/*"
+        agents:
+          slurm_mem: 100GB
```
Comment (charleskawczynski) [inline .buildkite/pipeline.yml:850]: I don't think this is a good idea

Comment (nefrathenrici) [reply .buildkite/pipeline.yml:850]: I agree, it was the only way I could get the job to complete. Maybe we should hold off on adding this perf test until I figure out what's going on with the ifelse allocations?

**Pull request merged**